### DISCUSSION TITLE
Expose digest function.

### DISCUSSION THREAD

**Discussion by jk89** [panva/jose#144] [Ideas]
Could you expose the runtime invarient DigestFunction via some utilities export? It's really useful, and I keep having to reimplement exactly as it appears in this library elsewhere.

**Comment by panva** [comment]
I'm not a fan of exposing internals. This is a JOSE library, not a general purpose universal crypto layer.

**Reply by jk89** [reply]
That is fair enough although you do expose some utilities like random which is also a primative IMO. I suppose you wouldn't mind me factorising it out into a little repo and publishing that? Props to your library, its very well written.

**Reply by panva** [reply]
I would not. It's MIT, do as you wish. Thank you for the kind words.
